tests/lava/proc/sparse/test_models.py

```diff
@@ -0,0 +1,588 @@
+# Copyright (C) 2021-22 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+import unittest
+import numpy as np
+
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.model.py.ports import PyOutPort, PyInPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.process.ports.ports import OutPort, InPort
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.resources import CPU
+from lava.magma.core.run_configs import RunConfig
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.proc.sparse.process import Sparse
+from lava.utils.weightutils import SignMode
+
+
+class SparseRunConfig(RunConfig):
+    """Run configuration selects appropriate Sparse ProcessModel based on tag:
+    floating point precision or Loihi bit-accurate fixed-point precision"""
+
+    def __init__(self, custom_sync_domains=None, select_tag="fixed_pt"):
+        super().__init__(custom_sync_domains=custom_sync_domains)
+        self.select_tag = select_tag
+
+    def select(self, proc, proc_models):
+        for pm in proc_models:
+            if self.select_tag in pm.tags:
+                return pm
+        raise AssertionError("No legal ProcessModel found.")
+
+
+class VecSendandRecvProcess(AbstractProcess):
```

Comment:
Can't we use existing `lava.proc.io` processes to do this and `VecRecvProcess`? If yes, we should do it and reuse existing code as much as possible.

Reply:
In that case the tests of the dense process should also change accordingly. I just duplicated the tests from dense since the sparse process should do and therefore be tested for the same thing and I had to upload a working implementation quickly

Reply:
If they can are different from `lava.proc.io` then, you can import them from `tests.lava.proc.dense.test_models`. Otherwise use the standard Io processes and maybe file an issue for Dense.